packages/flutter_tools/lib/src/commands/build_bundle.dart

```diff
@@ -81,6 +87,14 @@ class BuildBundleCommand extends BuildSubCommand {
     };
   }
 
+  @override
+  Future<void> validateCommand() async {
+    if (argResults['tree-shake-icons'] as bool) {
+      throwToolExit('The "--tree-shake-icons" flag is deprecated and will be removed in a future version of Flutter.');
```

Comment:
```suggestion
      throwToolExit('The "--tree-shake-icons" flag is deprecated for "build bundle" and will be removed in a future version of Flutter.');
```

Just so people don't think we're removing it elsewhere